client/features/data-encryption.js

```diff
@@ -0,0 +1,49 @@
+import WebSocketAsPromised from 'websocket-as-promised';
+
+export const decryptEventPayloads = async (events, port) => {
+  const sock = new WebSocketAsPromised(`ws://localhost:${port}/`, {
+    packMessage: data => JSON.stringify(data),
+    unpackMessage: data => JSON.parse(data),
+    attachRequestId: (data, requestId) =>
+      Object.assign({ requestId: requestId }, data),
+    extractRequestId: data => data && data.requestId,
+  });
+
+  try {
+    await sock.open();
+    const requests = [];
+
+    events.forEach(event => {
+      let payloads = [];
+
+      if (event.details.input) {
+        payloads = event.details.input.payloads;
+      }
+
+      if (event.details.result) {
+        payloads = event.details.result.payloads;
+      }
+
+      payloads.forEach((payload, i) => {
+        requests.push(
+          sock
+            .sendRequest({ payload: JSON.stringify(payload) })
+            .then(response => {
+              payloads[i] = JSON.parse(response.content);
+            })
+        );
+      });
+    });
+    await Promise.all(requests);
+  } catch (error) {
+    console.log(error);
+
+    return Promise.reject(error);
```

Comment:
what is the goal of this vs letting the stack trace show the source of the error?

suggestion - either add some extra info to the console.log to help with debugging, or just let the error throw and display nicely

Reply:
good point

I've added additional information to the error message so it would be easier to understand what has failed. Then added fallback logic to show original unprocessed payloads in case they fail to decrypt

![image](https://user-images.githubusercontent.com/11838981/115621858-efba3f80-a2ab-11eb-8884-17ba7d96e53b.png)